### PR TITLE
Deep links

### PR DESCRIPTION
**Pull request by skykelsey**

### CODE REVIEW

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/model/NavigateToLinkInteraction.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 public class NavigateToLinkInteraction extends Interaction {
 
 	private static final String KEY_URL = "url";
+	private static final String KEY_NEW_TASK = "new_task";
```
Comment (msaffitz) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/model/NavigateToLinkInteraction.java]: FYI: I added a comment to PLATFORM-47 about an idea on naming this.

**Comment by msaffitz** [comment]
Looks good, pending PLATFORM-47

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/model/NavigateToLinkInteraction.java [hunk 2]
```diff
@@ -26,4 +27,12 @@ public String getUrl() {
 		}
 		return null;
 	}
+
+	public boolean isNewTask() {
+		InteractionConfiguration configuration = getConfiguration();
+		if (configuration != null && !configuration.isNull(KEY_NEW_TASK)) {
+			return configuration.optBoolean(KEY_NEW_TASK, true);
```
Comment (pkamb) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/model/NavigateToLinkInteraction.java]: Default is `true` for opening in a new task. What does this mean for opening a link into your own app? Does that fail, open in a "new" task, or open in the same task?

**Comment by CodeMonkeySteve** [comment]
`target` is very vague, consider renaming to `target_task`.

**Comment by skykelsey** [comment]
@CodeMonkeySteve Can you follow up with Mike? If this changes again, I'd like it to be final.

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/NavigateToLinkInteractionView.java [hunk 3]
```diff
@@ -17,26 +19,39 @@
...
+				default:
+					break;
+			}
 
 			activity.startActivity(intent);
+			success = true;
+		} catch (ActivityNotFoundException e) {
+			Log.w("NavigateToLink Error: ", e);
 		} finally {
+			String data = String.format("{\"url\":\"%s\",\"target\":\"%s\",\"success\":%b}", interaction.getUrl(), interaction.getTarget().lowercaseName(), success);
```
Comment (msaffitz) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/NavigateToLinkInteractionView.java]: do you worry about anything potentially being in the url that needs escaping before going into JSON?

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java [hunk 4]
```diff
@@ -107,11 +116,14 @@ public void onClick(View view) {
 								if (interactionIdToLaunch != null) {
 									Interactions interactions = InteractionManager.getInteractions(activity);
 									if (interactions != null) {
-										Interaction interaction = interactions.getInteraction(interactionIdToLaunch);
+										Interaction invokedInteraction = interactions.getInteraction(interactionIdToLaunch);
 										if (interaction != null) {
```
Comment (msaffitz) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java]: should this be `invokedInteraction`?

File: samples/apptentive-dev/assets/interactions/TextModal Missing Body.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "type": "TextModal",
+  "name": "Centered TextModal Interaction",
+  "id": "textmodal_interaction_centered",
+  "priority": 1,
+  "configuration": {
+    "title": "Centered TextModal Title",
+    "actions": [
+      {
+        "label": "OnNE",
```
Comment (msaffitz) [inline samples/apptentive-dev/assets/interactions/TextModal Missing Body.json]: is this typo on purpose?

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/NavigateToLinkInteractionView.java (hunk 3 above)
Comment (skykelsey) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/NavigateToLinkInteractionView.java]: Good point.

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java (hunk 4 above)
Comment (skykelsey) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java]: Yes.

File: samples/apptentive-dev/assets/interactions/TextModal Missing Body.json (hunk 5 above)
Comment (skykelsey) [inline samples/apptentive-dev/assets/interactions/TextModal Missing Body.json]: Not on purpose, but I'll fix it.

**Comment by msaffitz** [comment]
:+1:

File: apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java (hunk 4 above)
Comment (pkamb) [inline apptentive/src/com/apptentive/android/sdk/module/engagement/interaction/view/TextModalInteractionView.java]: `interactionToInvoke` or `interactionToLaunch` I would think... it isn't "invoked" until the end.

**Pull request merged**